backend/app/Repositories/EloquentSessionRepository.php

```diff
@@ -6,11 +6,35 @@
 
 use App\Repositories\Contracts\SessionRepository;
 use Illuminate\Support\Collection;
+use Illuminate\Support\Carbon;
+use App\Entities\Session;
+use App\Actions\Sessions\GetAvgSessionRequest;
+use App\Entities\Visitor;
 
 class EloquentSessionRepository implements SessionRepository
 {
     public function getCollection(): Collection
     {
         return collect([]);
     }
+
+    public function getAvgSession(GetAvgSessionRequest $request): int
```

Comment:
instead of passing Request it is better to pass another DTO, for example AverageSessionFilter, with methods getVisitorsId(), getStartDate(), getEndDate(). And conversion to timestamp you should do outside of repository either in request or AverageSessionFilter